File: 11-container-with-most-water/11.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        head = 0
        tail = len(height) - 1
        area = 0

        while head < tail:
            area = max(area, min(height[head], height[tail]) * (tail - head))
            if height[head] < height[tail]:
                head += 1
            else:
                tail -= 1

        return area

File: 11-container-with-most-water/11.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size() - 1;
        int area = 0;

        while (left < right) {
            if (height[left] < height[right]) {
                area = max(area, height[left] * (right - left));
                ++left;
            } else {
                area = max(area, height[right] * (right - left));
                --right;
            }
        }

        return area;
    }
};
